### DISCUSSION TITLE
Shutdown/Restart option for a vSAN cluster / vSAN Shutdown Cluster Wizard

### DISCUSSION THREAD

**Discussion by ralfgro** [vmware/govmomi#3509] [Community Help]
I have a govc script that take care about shutting down VM, putting hosts into maintenance mode and shutting host down. This works for traditional clusters, it also works for vSAN clusters. But as far as I know, that is not the best practice for vSAN. On a vSAN cluster you can right click in vSphere Web Client and use the Shutdown Cluster Wizard to do this properly.

https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vsan-monitoring.doc/GUID-0D24C5BA-0248-4E3A-ADA3-33FF551C78F5.html

I did not find this option for govc. Can I still trigger this somehow by govc? Same for the Restart Cluster option that is available with vSAN.

**Comment by dougm** [comment]
We don't have that option in govc currently. First would need to figure out which API(s) are used for this. If you find any info on that and/or are able to use Code Capture in vCenter, please include those details here.
